sdk/core/Azure.Core/src/Pipeline/Internal/Multipart/BufferedReadStream.cs

```diff
@@ -243,7 +244,9 @@ public async override Task<int> ReadAsync(byte[] buffer, int offset, int count,
                 return toCopy;
             }
 
+#pragma warning disable CA1835 // WriteAsync(Memory<>) overload is not available in all targets
```

Comment:
Curious: why suppress this warning?

Reply:
It's done in a [few other places](https://grep.app/search?q=%23pragma%20warning%20disable%20CA1835&regexp=true&filter[repo][0]=Azure/azure-sdk-for-net&filter[path][0]=sdk/) - this is because the analyzer complains about not using the correct overload for the net6.0 target, but it's not available in the others. The analyzer didn't complain when it was in shared source.

Reply:
Is the analyzer telling us something important that we should pay attention to?  Why didn't it complain when it was in shared source?

In general, I think it's preferred for us to be paying down technical debt rather than increasing it as we move forward.

Reply:
This is net zero for current debt (as the code was simply relocated). I'd rather file an issue for the broader concern to investigate if this is worth changing everywhere it exists, rather than just here as part of this PR.

Reply:
created #29197